extensions/xcode/src/components/xcode-simulator-list/xcode-simulator-open-url-form.component.tsx

```diff
@@ -0,0 +1,40 @@
+import { XcodeSimulator } from "../../models/xcode-simulator/xcode-simulator.model";
+import { Action, ActionPanel, Clipboard, Form } from "@raycast/api";
+import { useEffect, useState } from "react";
+import { operationWithUserFeedback } from "../../shared/operation-with-user-feedback";
+import { XcodeSimulatorService } from "../../services/xcode-simulator.service";
+
+export function XcodeSimulatorOpenUrlForm(props: { simulator: XcodeSimulator }): JSX.Element {
+  const [url, setUrl] = useState<string>("");
+  useEffect(() => {
+    Clipboard.readText().then((url) => (url && XcodeSimulatorService.isValidUrl(url) ? setUrl(url) : undefined));
+  }, []);
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm
+            title="Open URL"
+            onSubmit={() =>
+              operationWithUserFeedback(
+                "Opening URL in Simulator",
+                "URL successfully opened in simulator",
+                "An error occurred while trying to open the url",
+                () => XcodeSimulatorService.openUrl(url, props.simulator)
```

Comment:
Couldn't we use the same error management as in the command so it's the same anywhere we use it?

Reply:
As the context is a little bit different in the `XcodeSimulatorOpenUrlForm` component we can simply ignore some errors for example:

- Xcode installation is guaranteed to be available as the `XcodeSimulatorList` is wrapped in a `XcodeInstallationVerifier` component.
- Missing booted simulator is not possible as the `Action.Push` to open the form is only available on a booted simulator.

Therefore wrapping the `XcodeSimulatorService.openUrl` in a `operationWithUserFeedback` should do the job.